### PR TITLE
feat: Custom domain resolution & template rules

### PR DESCRIPTION
**Pull request by ShaofeiZi**
Custom domain resolution & template rules
The default mode is not supported.
But you can leave a bit to let the user customize。

### CODE REVIEW

**Comment by kuitos** [comment]
Could you pls split the two features to two pr? Otherwise it is hard to review😀

**Comment by kuitos** [comment]
btw, I saw many changes with code formatting, could we revert these which not related with the feature?

**Comment by ShaofeiZi** [comment]
> Could you pls split the two features to two pr? Otherwise it is hard to review😀

Because the two functions are more relevant, I still tend to merge.

**Comment by ShaofeiZi** [comment]
> btw, I saw many changes with code formatting, could we revert these which not related with the feature?

TBR  Looks like the difference between tab and space.already edited.

File: src/index.js [hunk 1]
```diff
@@ -142,15 +142,15 @@ function execScripts(entry, scripts, proxy = window, opts = {}) {
...
+export default function importHTML(url, opts = {}) {
+	const { fetch = defaultFetch, getDomain = defaultGetDomain, customTemplateRules } = opts;
 	return embedHTMLCache[url] || (embedHTMLCache[url] = fetch(url)
 		.then(response => response.text())
 		.then(html => {
 
 			const domain = getDomain(url);
 			const assetPublicPath = `${domain}/`;
-			const { template, scripts, entry, styles } = processTpl(html, domain);
+			const { template, scripts, entry, styles } = processTpl(html, domain, customTemplateRules);
```
Comment (kuitos) [inline src/index.js]: 应该直接调用 customTemplateRules 然后把处理好的 html 传给 processTpl 就可以了，不用再修改 processTpl 的实现

File: index.d.ts [hunk 2]
```diff
@@ -17,10 +17,12 @@ interface IImportResult {
 
 type ImportEntryOpts = {
 	fetch?: Function;
+	getDomain?: Function;
+	customTemplateRules?: (string) => string;
```
Comment (kuitos) [inline index.d.ts]: 跟上面统一叫 getTemplate 好了

**Comment by kuitos** [comment]
thx for your contribution!

**Pull request merged**